docs/DeveloperGuide.md

```diff
@@ -285,29 +285,232 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+
+   Use case ends.
+
+**Extensions**
+
+* 1a. The command is invalid.
+    * 1a1. MediConnect shows an error message.
+
+      Use case resumes at step 1.
+
+
+**Use case: UC12 View a patient’s upcoming appointments**
+
+**MSS**
+
+1. User requests to view a patient’s upcoming appointments 
+2. MediConnect shows the patient’s appointment list
+
+   use case ends.
+
+**Extensions**
+
+* 1a. The command is invalid.
+    * 1a1. MediConnect shows an error message.
+
+      Use case resumes at step 1.
+
+* 1b. The patient does not exist in the system.
+    * 1b1. MediConnect shows an error message.
+    
+      Use case resumes at step 1.
+
+
+**Use case: UC13 View a doctor’s upcoming appointments**
+
+**MSS**
+
+1. User requests to view a doctor’s upcoming appointments. 
+2. MediConnect shows the doctor’s appointment list.
+
+   use case ends.
+
+**Extensions**
+
+* 1a. The command is invalid.
+    * 1a1. MediConnect shows an error message.
+
+      Use case resumes at step 1.
+
+* 1b. The doctor does not exist in the system.
+    * 1b1. MediConnect shows an error message.
+
+      Use case resumes at step 1.
+
+
+**Use case: UC14 Add an appointment between a patient and doctor**
+
+**MSS**
+
+1. User requests to add a specific appointment between a patient and doctor. 
+2. MediConnect adds the appointment to the list.
+
+   use case ends.
+
+**Extensions**
+
+* 1a.MediConnect detects that the arguments provided to add an appointment is invalid.
+    * 1a1. MediConnect shows an error message.
+    * 1a2 MediConnect requests for the correct input.
+      Use case resumes at step 1.
+
+* 1b. MediConnect detects that the appointment to be added already exists in the system
+    * 1b1. MediConnect shows an error message.
+
+      Use case ends.
+
+**Use case: UC15 Edit an appointment**
+
+**MSS**
+
+1. User requests to edit an appointment.
+2. MediConnect shows the current details of the appointment to the user.
+3. requests to edit the appointment. 
+4. MediConnect edits the doctor’s details.
 
     Use case ends.
 
 **Extensions**
 
-* 2a. The list is empty.
+* 1a.MediConnect detects that the arguments provided to edit an appointment is invalid.
+    * 1a1. MediConnect shows an error message.
+    * 1a2 MediConnect requests for the correct input.
 
-  Use case ends.
+      Use case resumes at step 1.
 
-* 3a. The given index is invalid.
+* 1b. MediConnect detects that the appointment to be edited does not exist in the system.
+    * 1b1. MediConnect shows an error message.
+  
+      Use case ends.
+* 3a.MediConnect detects that the arguments provided to edit an appointment is invalid.
+    * 3a1. MediConnect shows an error message.
+    * 3a2 MediConnect requests for the correct input
 
-    * 3a1. AddressBook shows an error message.
+      Use case resumes at step 3.
 
+**Use case: UC16 Delete an appointment**
+
+**MSS**
+
+1. User requests to list appointments 
+2. MediConnect shows a list of appointments
```

Comment:
@mounilsankar should listing be part of this use case? Whats the convention for this type of use cases do you know?

Reply:
It should not be part of the use case